src/cogent3/evolve/likelihood_tree.py

```diff
@@ -172,7 +172,11 @@ class LikelihoodTreeEdge(_LikelihoodTreeEdge):
 
     def sum_input_likelihoodsR(self, result, *likelihoods):
         return likelihood_tree.sum_input_likelihoods(
-            self.indexes, numpy.ascontiguousarray(result), likelihoods,
+            self.indexes
+            if self.indexes.flags["C_CONTIGUOUS"]
```

Comment:
put this conditional logic before the method call, makes the code easier to read.